packages/core/src/app/operator/block-db.ts

```diff
@@ -0,0 +1,146 @@
+/* External Imports */
+import BigNumber = require('bn.js')
+import { MerkleIntervalTree } from '@pigi/utils'
+
+/* Internal Imports */
+import { BaseKey } from '../common'
+import { KeyValueStore, BlockDB, StateUpdate } from '../../interfaces'
+
+const PREFIXES = {
+  VARS: new BaseKey('v').encode(),
+  BLOCKS: new BaseKey('b').encode(),
+}
+
+const KEYS = {
+  NEXT_BLOCK: Buffer.from('nextblock'),
+  BLOCK: new BaseKey('b', ['uint32']),
+}
+
+/**
+ * Simple BlockDB implementation.
+ */
+export class DefaultBlockDB implements BlockDB {
+  private vars: KeyValueStore
+  private blocks: KeyValueStore
+
+  /**
+   * Initializes the database wrapper.
+   * @param db Database to store values in.
+   */
+  constructor(private db: KeyValueStore) {
+    this.vars = this.db.bucket(PREFIXES.VARS)
+    this.blocks = this.db.bucket(PREFIXES.BLOCKS)
+  }
+
+  /**
+   * @returns the next plasma block number.
+   */
+  public async getNextBlockNumber(): Promise<number> {
+    // NOTE: You could theoretically cache the next block number as to avoid
+    // having to read from the database repeatedly. However, this introduces
+    // the need to ensure that the cached value and stored value never fall out
+    // of sync. It's probably better to hold off on implementing caching until
+    // this becomes a performance bottleneck.
+
+    const buf = await this.vars.get(KEYS.NEXT_BLOCK)
+    return buf.readUInt32BE(0)
+  }
+
+  /**
+   * Adds a state update to the list of updates to be published in the next
+   * plasma block.
+   * @param stateUpdate State update to publish in the next block.
+   * @returns a promise that resolves once the update has been added.
+   */
+  public async addPendingStateUpdate(stateUpdate: StateUpdate): Promise<void> {
+    const block = await this.getNextBlockStore()
+    const start = stateUpdate.id.start
+    const end = stateUpdate.id.end
+
+    // TODO: Figure out how to implement locking here so two state updates
+    // can't be added at the same time.
+
+    if (await block.has(start, end)) {
```

Comment:
How frequently do we imagine Operators will submit plasma blocks? There's still a lot for me to learn, but this line triggers some red flags around scalability, depending on the answer.

Reply:
Probably somewhere between 15sec and 10mins depending on the chain. Which part specifically raises red flags?

Reply:
Sorry @kfichter, just seeing this now, but it means that the same range can only be transacted once every 10-15sec. That's probably fine for now, but I can certainly think of applications for which that will be too slow (trading, poker, etc.).